File: a011(kindergarten_counting_game)/C++/kindergarten_counting_game.cpp
#include <cctype>
#include <iostream>
#include <string>


int main(int argc, char const *argv[]) {
  std::string sentence = "";

  while (std::getline(std::cin, sentence)) {
    /* add an extra non-alphabetic char to make sure the last word iss counted */
    sentence.push_back('-');
    
    int word_count = 0;
    bool in_word = false;
    for (const char c : sentence) {
      if (isalpha(c)) {
        in_word = true;
      } else {
        if (in_word) {
          ++word_count;
        }
        in_word = false;
      }
    }
    std::cout << word_count << '\n';
    sentence.clear();
  }
  return 0;
}

File: a011(kindergarten_counting_game)/Python/kindergarten_counting_game.py
from sys import stdin, stdout


for line in stdin:
    # add an extra non-alphabetic char to make sure the last word iss counted
    line += '-';

    in_word: bool = False
    word_count: int = 0
    for c in line:
        if c.isalpha():
            in_word = True
        else:
            if in_word:
                word_count += 1
            in_word = False
    stdout.write(f'{word_count}\n')
